contrib/platform/src/com/sun/jna/platform/mac/CoreFoundationUtil.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2019 Daniel Widdis
+ *
+ * The contents of this file is dual-licensed under 2
+ * alternative Open Source/Free licenses: LGPL 2.1 or later and
+ * Apache License 2.0. (starting with JNA version 4.0.0).
+ *
+ * You can freely decide which license you want to apply to
+ * the project.
+ *
+ * You may obtain a copy of the LGPL License at:
+ *
+ * http://www.gnu.org/licenses/licenses.html
+ *
+ * A copy is also included in the downloadable source code package
+ * containing JNA, in file "LGPL2.1".
+ *
+ * You may obtain a copy of the Apache License at:
+ *
+ * http://www.apache.org/licenses/
+ *
+ * A copy is also included in the downloadable source code package
+ * containing JNA, in file "AL2.0".
+ */
+package com.sun.jna.platform.mac;
+
+import static com.sun.jna.platform.mac.CoreFoundation.kCFStringEncodingUTF8;
+
+import java.util.Collection;
+
+import com.sun.jna.Memory;
+import com.sun.jna.platform.mac.CoreFoundation.CFAllocatorRef;
+import com.sun.jna.platform.mac.CoreFoundation.CFNumberType;
+import com.sun.jna.platform.mac.CoreFoundation.CFTypeRef;
+import com.sun.jna.ptr.IntByReference;
+import com.sun.jna.ptr.LongByReference;
+
+/**
+ * Provides utilities for Core Foundations
+ */
+public class CoreFoundationUtil {
+
+    private static final CoreFoundation CF = CoreFoundation.INSTANCE;
+
+    public static final CFAllocatorRef ALLOCATOR = CF.CFAllocatorGetDefault();
+
+    private CoreFoundationUtil() {
+    }
+
+    /**
+     * Convert a reference to a Core Foundations LongLong into its {@code long}
+     *
+     * @param theLong
+     *            The pointer to a 64-bit integer
+     * @return The corresponding {@code long}
+     */
+    public static long cfPointerToLong(CFTypeRef theLong) {
+        LongByReference lbr = new LongByReference();
+        CF.CFNumberGetValue(theLong, CFNumberType.kCFNumberLongLongType.ordinal(), lbr);
```

Comment:
According to the documentation, this operation can fail (https://developer.apple.com/documentation/corefoundation/1543114-cfnumbergetvalue) should the value be checked?

Reply:
With the `long` return type, even if we checked for failure we would have to return something (or throw an exception).  The remarks say it does attempt to return something though:
> If the argument type differs from the return type, and the conversion is lossy or the return value is out of range, then this function passes back an approximate value in valuePtr and returns false.

So I think the code is fine but perhaps the documentation should be improved to warn users of the Util about this potential.

Reply:
I expanded the javadoc comments for these methods. If you're still not comfortable with it, I'll remove it.  CFNumbers are weird, in that you are not guaranteed they will be stored in the same type that you created them, so you have to specify the type you want when you retrieve it, and it will attempt a conversion (and return an approximate value).

For example:
```
        DoubleByReference pi = new DoubleByReference(Math.PI);
        CFNumberRef cfPi = CF.CFNumberCreate(null, CFNumberType.kCFNumberDoubleType.ordinal(), pi);
        IntByReference ibr = new IntByReference();
        boolean success = CF.CFNumberGetValue(cfPi, CFNumberType.kCFNumberIntType.ordinal(), ibr);
        System.out.println("Success? " + success + ", value=" + ibr.getValue());
```
Returns:
> Success? false, value=3

We get the `false` because the double-to-int conversion was wrong, but we did get an approximate valued `int` which is what I want from the convenience method.

Reply:
I can live with it.

Reply:
OK, but having a day to think about it, I think I'll get rid of the Util class anyway.  Since the `CFStringRef` class already turns a String into a reference, it should also do the opposite, so this method belongs over there (as `stringValue()`).   The `CFBooleanRef` conversion isn't even useful (it's a one-liner) but I could add a `booleanValue()` method to that class. And I could move the `CFNumberRef` options into that class with `longValue()`, etc. and expand to the full list of `Number` types that the `Variant` class already handles similarly.